lib/workers/repository/config-migration/branch/create.ts

```diff
@@ -34,6 +34,7 @@ export async function createConfigMigrationBranch(
     migratedConfigData
   );
   return commitAndPush({
+    targetBranch: config.baseBranch,
```

Comment:
Do we still need those changes? or can we simply remove that again?

Reply:
For my Gerrit implementation i always need the target branch.
But i have still not tested a config-migration with my Gerrit implementation.
 
Why do you think this information is unnecessary in this case? 
Perhaps i don't understand the details about the process of config-migration.

Reply:
I don't think we need to pass `baseBranch` as `targetBranch` here, as the git layer already knows the `baseBranch`.
So way we need to pass it again? It's always the same.

Reply:
but not for the future Gerrit implementation!
I need this information to create a Gerrit-"PullRequest" (i.e. Gerrit-Change).
There is no remote branch available with the name `getMigrationBranchName(config)`.

The Gerrit implementation will create/update a change onto a target branch (main, master, next whatever the user has Chosen as baseBranch).

Reply:
You get this information when renovate is calling `git.checkoutBranch`. That should be the target branch where the PR should point to. 🤔

Reply:
What exactly do you mean? 
I should store the "current" effective target branch internally anywhere inside the Gerrit implementation from each checkout call?
That sounds like abuse, because nobody restricts the checkoutBranch function to only be used for checking out a baseBranch(TargetBranch).

Reply:
that's what the current git layer expect to happen. otherwise we would already pass that information. 🤔

Reply:
No, that is what the current supported git platforms are expecting. They push a new branch (name created from the dependency) to origin under the **same name**.

But for gerrit this will not work. I need the name of the base branch, the target of the pull request.

Reply:
But i've see the problem.
The (default) `commitFiles()` function has to work like 
```
git commit ...
git push origin $branchname:$branchname
```
It don't need the targetBranch as push target.
But the information needs to be passed down to allow the Gerrit specific `commitFiles` function to work like
```
git commit ...
git push origin $branchname:refs/for/$targetBranch  //$branchname will be HEAD
```
 
I will correct the (default) commitFiles function accordingly.

Perhaps we should rename the `targetBranch`  to `baseBranch` to make this clear?